src/server/server.ts

```diff
@@ -83,9 +88,19 @@ async function serveArticle(req: Request, res: ExpressResponse): Promise<void> {
             }
           },
           content => {
-            const { resources, element } = Page({ content, imageSalt, getAssetLocation });
+            const {
+              resources,
+              element,
+              hydrationProps
+            } = Page({ content, imageSalt, getAssetLocation });
+            const props = JSON.stringify({ ...hydrationProps, imageSalt })
+              .replace(/’/g, "&rsquo;")
+              .replace(/“/g, "&ldquo;")
+              .replace(/”/g, "&rdquo;")
+              .replace(/…/g, "&hellip;");
             res.set('Link', getPrefetchHeader(resources));
             res.write('<!DOCTYPE html>');
+            res.write(`<script>window.__INITIAL__DATA__ = ${props}</script>`)
```

Comment:
I think this will sidestep our bundling process and remove some of the value of our CSP. Could we achieve this another way? Either through additional requests or [`<meta>` tags](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta)?

Reply:
Any thoughts on this?

Reply:
I've tried some ways to use `meta` tags and also just plan JSON with script tags but couldn't find a way...

Reply:
Did you hit problems with `<script type="application/json">${props}</script>`?

Do we have to hydrate immediately? Since the UI won't change until the first poll request goes out (or until the user clicks "view more"), can we wait until the first request comes back to hydrate? That way we can perform a standard request for JSON and use the result to populate the body of the page.

Reply:
I'll try the script json again, I think my problem before was UTF-8 related. I think the additional network request would have to be a new mapi endpoint? and another request to capi?

Reply:
pushed a commit using `<script type="application/json">`